K readme

emcwhirter-usgs opened the pull request.

hasimpson-usgs commented:
Need to include how the 3 hour window is calculated, and what the windows are.   Especially since when we do a realtime product,  it may differ.

hasimpson-usgs commented:
Might want to add a Statistical smoothing section,  with a TODO line in it.

erigler commented on docs/K-USGS.md:
```diff
@@ -0,0 +1,127 @@
+calculated with just the raw time-series data as inputs.
+
+The USGS has been producing digital K-Indices since 1979. The K-USGS algorithm
+was one of many written to attempt to simulate the previously used hand-scaled
```
Add a citation to the Menvielle-1995 paper here, and a reference+link at the end. I know we're not really following that paper, but it is kind of the definitive validation paper for all these different algorithms.

erigler commented on docs/K-USGS.md:
```diff
@@ -0,0 +1,127 @@
+interval. Each observatory has an adopted scale, which is a multiple of the
+Niemegk observatory’s scale shown below. This scale is used to translate the K
+variation nano-Tesla (nT) value of the data onto a 0 to 9 integer scale. The
+“Range” represents this nT cutoff for each “KValue”.
```
I'm not really sure what is meant by "Range" and "cutoff" here. What should we do if the K-variation is 69nT at Niemegk, just set it to 4, or round up to 5? I honestly don't know the answer, but this seems like more than a trivial consideration, and should be addressed in the README.

erigler commented on docs/K-USGS.md:
```diff
@@ -0,0 +1,127 @@
+calendar day in the desired data range. In order to get a good representative
+curve of the magnetic activity, some data points should be excluded from
+calculations. These exclusions are:
+* MHVs containing minute values having an extreme range.
```
How should "extreme", and "tails of the distribution" be defined? I imagine these will be important runtime configuration parameters, and possibly different for each observatory.

erigler commented on docs/K-USGS.md:
```diff
@@ -0,0 +1,127 @@
+ 1. Eliminate any 1-Minute Values and Mean Hourly Values that fall outside of
+    acceptable statistical limits.
+ 2. Create daily SR-curves for every day in the selected data window.
+ 3. Remove the SR-curve to find the K variation of the data.
```
Replace "Remove" with "Subtract" to avoid ambiguity.

hasimpson-usgs replied:
Can you put these in the comment section of github?

On Thu, Mar 26, 2015 at 2:33 PM, erigler-usgs <notifications@github.com>
wrote:

> In docs/K-USGS.md
> <https://github.com/usgs/geomag-algorithms/pull/17#discussion_r27253046>:
>
> > +calculated with just the raw time-series data as inputs.
> > +
> > +The USGS has been producing digital K-Indices since 1979. The K-USGS algorithm
> > +was one of many written to attempt to simulate the previously used hand-scaled
> > +process as closely as possible, and was one of the four to be officially
> > +accepted by an IAGA Working Group V during the Vienna IUGG General Assembly in
> > +1991. This method is described in "[Wilson (1987)](#wilson-1987). The algorithm
> > +was tested and found acceptable in 1990 using IAGA WG-5 data. It was tested
> > +again in 2010 using a subset of USGS data, and was found to still be acceptable
> > +for producing K-Indices.
> > +
> > +In general, the K-USGS algorithm consists of these steps:
> > + 1. Eliminate any 1-Minute Values and Mean Hourly Values that fall outside of
> > +    acceptable statistical limits.
> > + 2. Create daily SR-curves for every day in the selected data window.
> > + 3. Remove the SR-curve to find the K variation of the data.
>
> Replace "Remove" with "Subtract" to avoid ambiguity.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/usgs/geomag-algorithms/pull/17/files#r27253046>.
>



-- 
Hal Simpson
IT Specialist
USGS Geomagnetism Program
Office:303-273-8459
Email:hasimpson@usgs.gov

erigler commented:
The last paragraph Section 3.1 of the Wilson-1987 paper is:

>Each set of MHVs used to estimate an SR curve also include the last two MHVs of the previous day and the first two hours of the next day to maintain SR curve continuity. If the contiguous MHVs are absent the daily or monthly mean is used. The additional data also improves the SR curve fit when the maximum diurnal effect occurs at approximately 2404 Universal time.

This is where I got my idea that monthly means are used to replace bad MHVs. I'm not actually convinced my interpretation is correct, especially in light of looking at some of the MagProc code, but how *should* we interpret this paragraph?

erigler commented:
[Moving line notes to comments section]

How should "extreme", and "tails of the distribution" be defined when filtering MHVs? I imagine these will be important runtime configuration parameters, and possibly different for each observatory.

erigler commented:
[Moving line notes to comments section]

Replace "Remove the SR curve" with "Subtract the SR curve" to avoid ambiguity in the general K-USGS algorithm description.

erigler commented:
[Moving line notes to comments section]

I'm not really sure what is meant by "Range" and "cutoff" in the Translate section, just before the Niemegk table. What should we do if the K-variation is 69nT at Niemegk, just set it to 4, or round up to 5? I honestly don't know the answer, but this seems like more than a trivial consideration, and should be addressed in the README.

erigler commented:
Add a citation to the Menvielle-1995 paper near "The K-USGS algorithm was one of many written ...", and of course a reference+link at the end. I know we're not really following that paper, but it is kind of the definitive validation paper for all these different algorithms.

erigler commented:
Hal,

Just to be clear, this document is not intended for real time K indices. In fact, at some level, it is impossible to do a real time K in the same spirit as Bartels and Mayaud. If/when we get around to (re)doing a "Real Time K", I'm pretty sure the algorithm will look very little like this one.

emcwhirter-usgs commented:
Added citation to Menvielle 1995 paper.
"Subtract" instead of "Remove" the SR-Curve.
Attempted to clarify 3-hour window and "Range" in Translate section.

"Extremes" and "tails of the distribution" are things we can/will define as we tune the algorithm; I added TODO statments to remind us to fill these in as we figure them out.

I agree that MHVs should be replace with something, and that something is stated in Wilson's papers, so I expanded the "Clean the Data" section to include the replacement of MHVs that are removed.
